scss/_layouts_fluid-breakout.scss

```diff
@@ -0,0 +1,216 @@
+  }
+
+  %l-fluid-breakout__aside--right-ie-11-fallback#{$suffix} {
+    @extend %l-fluid-breakout__aside-common-properties-ie-11-fallback#{$suffix};
+
+    @media (max-width: $l-fluid-breakout-3-col-threshold) {
+      padding-left: 0;
+    }
+  }
+
+  .l-fluid-breakout#{$suffix} {
+    @extend %l-fluid-breakout-ie11-fallback#{$suffix};
+
+    @supports (display: grid) {
+      display: block;
+      grid-gap: 0 0;
+    }
+
+    @media (min-width: $breakpoint-large) {
+      margin-left: auto;
+      margin-right: auto;
+      max-width: 2 * $l-fluid-breakout-aside-width + $l-fluid-breakout-max-width;
+
+      @supports (display: grid) {
+        display: grid;
+        grid-template-columns:
+          minmax($l-fluid-breakout-aside-width, 1fr)
+          minmax(0, $l-fluid-breakout-max-width)
+          minmax($l-fluid-breakout-aside-width, 1fr);
+        grid-template-rows: auto;
+      }
+    }
+
+    @media (max-width: $threshold-4-6-col) {
+      padding-left: map-get($grid-margin-widths, small);
+      padding-right: map-get($grid-margin-widths, small);
+    }
+
+    @media (min-width: $threshold-4-6-col) and (max-width: $threshold-6-12-col) {
+      padding-left: map-get($grid-margin-widths, medium);
+      padding-right: map-get($grid-margin-widths, medium);
+    }
+
+    @media (min-width: $threshold-6-12-col) and (max-width: $l-fluid-breakout-3-col-threshold) {
+      padding-left: map-get($grid-margin-widths, large);
+      padding-right: map-get($grid-margin-widths, large);
+    }
+
+    @media (min-width: $l-fluid-breakout-3-col-threshold) {
+      padding-left: 0;
+      padding-right: 0;
+
+      @supports (display: grid) {
+        grid-gap: 0 0;
+      }
+    }
+
+    %l-fluid-breakout__full-and-main-item-common#{$suffix} {
+      @extend %l-fluid-breakout__full-and-main-item-ie11-fallback#{$suffix};
+
+      @supports (display: grid) {
+        // overwrite fallback margins
+        margin-right: 0;
+      }
+    }
+
+    .l-fluid-breakout__full-item {
+      @extend %l-fluid-breakout__full-and-main-item-common#{$suffix};
+    }
+
+    .l-fluid-breakout__main-item {
+      @extend %l-fluid-breakout__full-and-main-item-common#{$suffix};
+
+      @media (min-width: $l-fluid-breakout-3-col-threshold) {
+        @supports (display: grid) {
+          grid-column: auto / auto;
+          width: initial;
+        }
+      }
+    }
+
+    %l-fluid-breakout__main-and-full-common#{$suffix} {
+      @extend %l-fluid-breakout__full-and-main-ie11-fallback#{$suffix};
+
+      @supports (display: grid) {
+        display: grid;
+        grid-gap: 0 $grid-gap;
+        grid-row: 1;
+        grid-template-columns: repeat(auto-fit, minmax($l-fluid-breakout-main-child-width, 1fr));
+        width: 100%;
+      }
+
+      @media (min-width: $l-fluid-breakout-3-col-threshold) {
+        padding-left: map-get($grid-margin-widths, large);
+        padding-right: map-get($grid-margin-widths, large);
+      }
+    }
+
+    .l-fluid-breakout__full {
```

Comment:
Do we need to have a separate class name for `__full`? "Full" sounds like a modifier, not an element. It seems that it's just a variant of `__main`. So maybe we could have it as `l-fluid-breakout__main.is-full-width`?

I think with layouts we should aim for a common structure of HTML and modify the layout by modifier classes, not by changing class names on children.

So in this case I think our default structure should be:

```html
<div class="l-fluid-breakout">
   <div class="l-fluid-breakout__aside"></div>
   <div class="l-fluid-breakout__main">
       <div class="l-fluid-breakout__item">
   </div>
</div>
```

Where `__aside` is optional (and can be before or after).
And `__main` is always there, but can be modified. If there is no aside, you add `.no-aside` to it.
If you want it to take full width you add `is-full-width` (instead of replacing it with `__full`).

Also, for an "item" class name we should use one consistent `l-fluid-breakout__item`, and not distinguish between `main-item` and `full-item`. From HTML point of view they are the same, we should be able to change their style based on parent modifier.

Would that be possible?